### PR TITLE
WIP: Adding Storage Unit Test App

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
This is not ready for review.

I am not able to use Firebase in the new test app that I created (test_apps/firebase_storage_unit_test_app). I am getting the following error:

"Default FirebaseApp failed to initialize because no default options were found. This usually means that com.google.gms:google-services was not applied to your gradle project."

I believe that "test_apps/build.gradle" is not included in the :test-apps:firebase_storage_unit_test_app build. When I add `apply com.google.gms:google-services` to the firebase_storage_unit_test_app.gradle, AS complains that it doesn't know about the plugin, even though `test-apps/build.gradle` adds it to the classpath.

### CODE REVIEW

**Comment by schmidt-sebastian** [comment]
@vkryachko Do you have a suggestion that might help here?

**Comment by vkryachko** [comment]
@schmidt-sebastian If you are referring to CI failures:
1. You need to format source code in both storage and test-apps/storage with google-java-format
2. Add license header to `ConnectionInjector.java`
3. Ignore smoke tests, @allisonbm92 was going to disable them as they are really flaky

File: subprojects.cfg [hunk 1]
```diff
@@ -10,10 +10,15 @@ fiamui-app
...
 transport:transport-api
 transport:transport-runtime
 transport:transport-backend-cct
 
 
 tools:apksize
 tools:errorprone
+
+
+test-apps
```
Comment (vkryachko) [inline subprojects.cfg]: this needs to go as well

File: test-apps/storage-unit-test-app/storage-unit-test-app.gradle [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+apply plugin: "com.android.application"
+
+android {
+  compileSdkVersion project.targetSdkVersion
```
Comment (vkryachko) [inline test-apps/storage-unit-test-app/storage-unit-test-app.gradle]: can you add `testBuildType = project.testBuildType` here? it's needed to respect smoke-tests parameters

File: test-apps/storage-unit-test-app/storage-unit-test-app.gradle [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  androidTestImplementation 'com.android.support.test:rules:1.0.2'
+}
+
+
+// ==========================================================================
+// Copy from here down if you want to use the google-services plugin in your
+// androidTest integration tests.
+// ==========================================================================
+ext.packageName = "com.example.flubber"
+apply from: '../../gradle/googleServices.gradle'
```
Comment (vkryachko) [inline test-apps/storage-unit-test-app/storage-unit-test-app.gradle]: This is not needed, google-services.json is automatically handled for smoke tests, see `test-apps/build.gradle` for hot it happens.

**Pull request closed without merging**